scalafix/rules/src/main/scala/scalafix/internal/rule/RscCompat.scala

```diff
@@ -0,0 +1,537 @@
+// Copyright (c) 2017-2018 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0 (see LICENSE.md).
+// NOTE: This file has been partially copy/pasted from scalacenter/scalafix.
+package scalafix.internal.rule
+
+import java.io._
+import java.nio.charset.StandardCharsets.UTF_8
+import scala.collection.mutable
+import scala.meta._
+import scala.meta.contrib._
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.semanticdb.Scala.{Descriptor => d}
+import scala.meta.internal.semanticdb.SymbolInformation.{Kind => k}
+import scala.meta.internal.semanticdb.SymbolInformation.{Property => p}
+import scalafix.internal.patch.DocSemanticdbIndex
+import scalafix.internal.util._
+import scalafix.lint.LintMessage
+import scalafix.rule._
+import scalafix.syntax._
+import scalafix.util.TokenOps
+import scalafix.v0._
+
+// ============ Introduction ============
+//
+// RscCompat is a Scalafix rule that rewrites Scala codebases
+// to be compatible with Rsc. At the moment, it is far from perfect -
+// for details, see https://github.com/twitter/rsc/labels/Scalafix.
+
+case class RscCompat(index: SemanticdbIndex) extends SemanticRule(index, "RscCompat") {
+  override def fix(ctx: RuleCtx): Patch = {
+    val targets = collectRewriteTargets(ctx.tree)
+    targets.map(ascribeReturnType(ctx, _)).asPatch
+  }
+
+  private case class RewriteTarget(env: Env, name: Name, body: Term)
+
+  private def collectRewriteTargets(tree: Tree): List[RewriteTarget] = {
+    val buf = List.newBuilder[RewriteTarget]
+    def append(env: Env, name: Name, body: Term): Unit = {
+      buf += RewriteTarget(env, name, body)
+    }
+    def loop(env: Env, tree: Tree): Unit = {
+      // FIXME: https://github.com/twitter/rsc/issues/140
+      tree match {
+        case Source(stats) =>
+          stats.foreach(loop(env, _))
+        case Pkg(_, stats) =>
+          stats.foreach(loop(env, _))
+        case Pkg.Object(_, name, templ) =>
+          loop(TemplateScope(name.symbol.get.syntax) :: env, templ)
+        case Defn.Class(_, name, _, _, templ) =>
+          loop(TemplateScope(name.symbol.get.syntax) :: env, templ)
+        case Defn.Trait(_, name, _, _, templ) =>
+          loop(TemplateScope(name.symbol.get.syntax) :: env, templ)
+        case Defn.Object(_, name, templ) =>
+          loop(TemplateScope(name.symbol.get.syntax) :: env, templ)
+        case Template(early, _, _, stats) =>
+          (early ++ stats).foreach(loop(env, _))
+        case Defn.Val(_, List(Pat.Var(name)), None, body) =>
+          append(env, name, body)
+        case Defn.Var(_, List(Pat.Var(name)), None, Some(body)) =>
```

Comment:
I'm not completely sure about what exactly needs to be done in RscCompat, so I'd like to avoid extra noise on the console for the time being. However, I've introduced a fixme comment and added the corresponding ticket to the current milestone.